src/geolocation.re

```diff
@@ -0,0 +1,104 @@
+type rnConfig;
+
+type currentPositionConfig;
+
+type watchPositionConfig;
+
+type watchId = int;
+
+type position = {
+  .
+  "coords": coords,
+  "timestamp": float,
+}
+and coords = {
+  .
+  "speed": int,
+  "longitude": float,
+  "latitude": float,
+  "accuracy": int,
+  "heading": int,
+};
+
+type error = {
+  .
+  "code": int,
+  "message": string,
+};
+
+[@bs.obj]
+external makeRnConfig : (~skipPermissionRequests: Js.boolean=?) => rnConfig =
+  "";
+
+[@bs.obj]
+external makeCurrentPositionConfig :
+  (~timeout: int=?, ~maximumAge: int=?, ~enableHighAccuracy: Js.boolean=?) =>
+  currentPositionConfig =
+  "";
+
+[@bs.obj]
+external makeWatchPositionConfig :
+  (
+    ~timeout: int=?,
+    ~maximumAge: int=?,
+    ~enableHighAccuracy: Js.boolean=?,
+    ~distanceFilter: int=?,
+    ~useSignificantChanges: Js.boolean=?
+  ) =>
+  watchPositionConfig =
+  "";
+
+[@bs.val "navigator.geolocation.setRNConfiguration"]
+external _setRNConfiguration : rnConfig => unit = "";
+
+let setRNConfiguration = (~skipPermissionRequests=?, ()) =>
+  _setRNConfiguration(makeRnConfig(~skipPermissionRequests?));
+
+[@bs.val "navigator.geolocation.requestAuthorization"]
+external requestAuthorization : unit => unit = "";
+
+[@bs.val "navigator.geolocation.stopObserving"]
+external stopObserving : unit => unit = "";
+
+[@bs.val "navigator.geolocation.getCurrentPosition"]
+external _getCurrentPosition :
```

Comment:
These externals don't need to have the _ prefix. If you create a let binding directly after, this won't be accessible only the let binding.

Reply:
this is how the navigator global looks like:

<img width="806" alt="react native debugger 2018-03-23 19-37-51" src="https://user-images.githubusercontent.com/1467712/37847640-d0323c68-2ed1-11e8-90d3-ab711020fdb4.png">

let's create `navigator.re` module, and add the `Geolocation` module there
we could add other bindings from the `navigator` global in next pull requests, what do you think about that? 😉